### PR TITLE
perf(cmp-cmdline): lazy load

### PR DESCRIPTION
**Pull request by meijieru**
## 📑 Description

Lazy load `cmp-cmdline`, for faster startup time.

## ℹ Additional Information

Original implementation load `cmp` at startup.

### CODE REVIEW

File: lua/astrocommunity/completion/cmp-cmdline/init.lua [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+    },
+    config = function(_, opts) require("cmp").setup.cmdline(":", opts) end,
+  },
+  {
+    "nvim-cmp",
+    event = { "CmdlineEnter" },
+    dependencies = {
+      "cmp-cmdline",
+    },
   },
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/cmp-cmdline/init.lua:30]: Why do you have this?

Comment (meijieru) [reply lua/astrocommunity/completion/cmp-cmdline/init.lua:30]: Otherwise, the `cmp-cmdline` won't get loaded.

Comment (Uzaaft) [reply lua/astrocommunity/completion/cmp-cmdline/init.lua:30]: I believe that's the incorrect way to do it.
Instead, you should make `nvim-cmp` a dependency of `cmp-cmdline`, and make `cmp-cmdline` load on CmdLineEnter.

Comment (meijieru) [reply lua/astrocommunity/completion/cmp-cmdline/init.lua]: fixed.

File: lua/astrocommunity/completion/cmp-cmdline/init.lua [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
+      mapping = cmp_mapping.preset.cmdline(),
+      sources = cmp_sources({
+        { name = "path" },
+      }, {
+        {
+          name = "cmdline",
+          option = {
+            ignore_cmds = { "Man", "!" },
+          },
         },
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/cmp-cmdline/init.lua:17]: Any particular reason why you changed the opts to a function?

Comment (meijieru) [reply lua/astrocommunity/completion/cmp-cmdline/init.lua:17]: That's for further lazy loading the `cmp_sources` & `cmp_mappings`.

**Review by Uzaaft — APPROVED** [review]

**Pull request merged**